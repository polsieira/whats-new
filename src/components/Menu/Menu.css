.Menu {
  color: #ffffff;
}

.header--yellow {
  color: #FFAE02;
}