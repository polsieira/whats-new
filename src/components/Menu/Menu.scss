.Menu {
  color: #ffffff;
  width: 20%;
}

.header {
  margin: 15px 0px;
}

.header--yellow {
  color: #FFAE02;
}

ul {
  list-style-type: none;
}

li {
  border-radius: 0px 30px 30px 0px; 
  cursor: pointer;
  font-size: 26px;
  margin-right: 5px;
  padding: 20px 0px 20px 10px;
  text-align: left;
  &:hover {
    background: #FFAE02;
  }
}