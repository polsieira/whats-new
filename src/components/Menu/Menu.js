import React from 'react';
import { IoIosPin } from "react-icons/io";
import './Menu.scss';

const Menu = ({ changeState }) => {
  return (
    <aside className='Menu'>
      <h1 className="header">What's <span className="header--yellow">New?</span></h1>
      <ul>
        <li id="local" onClick={(event) => changeState(event)}><IoIosPin size={30} />Local News</li>
        <li id="technology" onClick={(event) => changeState(event)}>Technology</li>
        <li id="entertainment" onClick={(event) => changeState(event)}>Entertainment</li>
        <li id="science" onClick={(event) => changeState(event)}>Science</li>
        <li id="health" onClick={(event) => changeState(event)}> Health</li>
      </ul>
    </aside >
  )
}

export default Menu;