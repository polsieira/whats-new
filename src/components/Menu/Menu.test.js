import React from 'react';
import { shallow } from 'enzyme';
import Menu from './Menu';

describe('Menu', () => {
  let wrapper;
  const mockChangeState = jest.fn();
  beforeEach(() => {
    wrapper = shallow(<Menu
      changeState={mockChangeState} />);
  })

  it('should match the snapshot with all data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('should call change state when any category is clicked', () => {
    it('category: local', () => {
      //Setup
      const mockEvent = {
        target: {
          id: 'local'
        }
      };

      //Execution
      wrapper.find('li').at(0).simulate('click', mockEvent);

      //Expectation
      expect(mockChangeState).toHaveBeenCalledWith(mockEvent);
    });
  })
});