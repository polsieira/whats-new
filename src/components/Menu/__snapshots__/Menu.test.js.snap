// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should match the snapshot with all data passed in correctly 1`] = `
<aside
  className="Menu"
>
  <h1
    className="header"
  >
    What's 
    <span
      className="header--yellow"
    >
      New?
    </span>
  </h1>
  <ul>
    <li
      id="local"
      onClick={[Function]}
    >
      <IoIosPin
        size={30}
      />
       Local News
    </li>
    <li
      id="technology"
      onClick={[Function]}
    >
      <IoMdCog
        size={30}
      />
       Technology
    </li>
    <li
      id="entertainment"
      onClick={[Function]}
    >
      <IoIosMusicalNotes
        size={30}
      />
       Entertainment
    </li>
    <li
      id="science"
      onClick={[Function]}
    >
      <IoIosFlask
        size={30}
      />
       Science
    </li>
    <li
      id="health"
      onClick={[Function]}
    >
      <IoIosMedkit
        size={30}
      />
       Health
    </li>
  </ul>
</aside>
`;
